modules/gapi/cmake/detectOneVPL.cmake

```diff
@@ -0,0 +1,34 @@
+if(WITH_GAPI_ONEVPL)
+    if (NOT ONEVPL_DIR)
+        message("Search oneVPL package by PATH")
+        find_package(VPL QUIET)
```

Comment:
`ONEVPL_DIR` may be unused in `find_package()`. There are tons of other methods for searching packages. We shuold not stick to this one.
Refer to CMake documentation.

Reply:
definitely. is it unused here in this part of branch but has worth in `else()` branch for searching VPL in specific location

The idea behind this implementation is the following:
**I)** If `-DWITH_ONEVPL` is set ON **then** we tried to search it in default PATH **and** used found one **or** downloaded new from github

It works when:  
a) user sourced `vars.sh` from oneVPL package before (Normal way as oneAPI usage suggested). In this case **we do not need** to find_package, just for sanity check
b) oneVPL installed by default SYSTEM PATH. In this case `find_package` do not use `ONEVPL_DIR` as well
c) If no one found, then we clone it from git hub using `ExternalProjectAdd`

**II)** another option is passing `ONEVPL_DIR` as external oneVPL package location:
a) In this case we force `find_package` to find oneVPL by `ONEVPL_DIR` only (do not check PATH). It's `else()` branch for this `if-condition`. I suggest  this options for "Custom-oneVPL" package to work-around issues which are not fixed in main oneVPL repo or official release package or some customization


If I understood your previous comment about `ExternalProject_Add` correctly then you suggest to eliminate "download&build" from github feature and we should retain **I-a,b** and **II-a**

But I find "download&build"  very usefull at least for current development :) (May we wrapped it by additional FORCE_DOWNLOAD_ONEVPL flag ?)

Reply:
No download. No check for `ONEVPL_DIR`.
Just `find_package(VPL)`

Reply:
removed. just bare `find_package` is used at now